src/logic/hooks/useEstimationStatus.tsx

```diff
@@ -0,0 +1,32 @@
+import { Dispatch, SetStateAction, useEffect, useState } from 'react'
+
+import { EstimationStatus } from './useEstimateTransactionGas'
+import { ButtonStatus } from 'src/components/Modal'
+
+export const useEstimationStatus = (
+  txEstimationStatus?: EstimationStatus,
+): [buttonStatus: ButtonStatus, setButtonStatus: Dispatch<SetStateAction<ButtonStatus>>] => {
+  const [buttonStatus, setButtonStatus] = useState<ButtonStatus>(ButtonStatus.DISABLED)
+
+  useEffect(() => {
+    let mounted = true
+
+    if (mounted) {
+      switch (txEstimationStatus) {
+        case EstimationStatus.LOADING:
+          setButtonStatus(ButtonStatus.LOADING)
+          break
+        case EstimationStatus.FAILURE:
```

Comment:
default perhaps, instead of FAILURE and SUCCESS, if something went wrong, it will leave the button disabled for good